### ISSUE TITLE
Cluster and load balancer name are not getting applied via transform

### ISSUE THREAD

**Issue by imdkbj** [sst/ion#633]
**Cluster**

Expected : dev-cluster
Actual :  abc-dev-**FargateCluster**

```js
    // Create an ECS cluster
    const cluster = new sst.aws.Cluster("Fargate", {
      vpc,
      transform: {
        cluster: {
          name: `${_stage}-cluster`,
          settings: [
            {
              name: "containerInsights",
              value: "enabled"
            }
          ]
        }
      }
    });
 ```
    
 **LB**
Expected : dev-abc
Actual :  dev-**AbcTask**

```js
 transform: {
        loadBalancer: {
          name:`${_stage}-${_name}`;
        },
      }

**Comment by jayair** [comment]
Hmm it's strange that `dev-cluster` is turning into `abc-dev-FargateCluster`. I might be missing something but what is `abc` here?

**Comment by imdkbj** [comment]
Hey, Thaks for the reply and sorry fot not putting the same. abc is the app name here.

**Comment by imdkbj** [comment]
Ah! Same naming issue found for lambda too.

<img width="795" alt="image" src="https://github.com/user-attachments/assets/c1d8e9ef-0719-449b-8dbe-182d7ff79199">

**Comment by shmuli9** [comment]
I have noticed a similar issue for Postgres resource

```typescript
  const db =
    new sst.aws.Postgres("DB", {
      databaseName,
      transform: {
        instance: {
          publiclyAccessible: true,
          identifier: "app-staging-db-instance",
        },
        cluster: {
          clusterIdentifier: "app-staging-db-cluster",
        },
      },
    });
```

But in AWS Console, the identifier is showiing as `app-main-dbcluster` where main is the name of the stage

**Comment by shmuli9** [comment]
I believe this [codeblock](https://github.com/sst/ion/blob/dev/platform/src/components/component.ts#L66-L232) is the issue

@fwang Why are these values overridden in this way?

**Comment by fwang** [comment]
@shmuli9 i was just dumb

Fixed in v0.1.44.

**Comment by shmuli9** [comment]
@fwang, thanks bro. Appreciate you maintaining this project
